ibis/backends/dask/udf.py

```diff
@@ -0,0 +1,121 @@
+import itertools
+
+import dask.dataframe as dd
+import dask.dataframe.groupby as ddgb
+import pandas
+
+import ibis.client
+import ibis.expr.operations as ops
+import ibis.udf.vectorized
+from ibis.backends.pandas.client import ibis_dtype_to_pandas
+from ibis.backends.pandas.udf import nullable, udf  # noqa
+
+from .dispatch import execute_node, pre_execute
+from .execution.util import assert_identical_grouping_keys, make_selected_obj
+
+
+@pre_execute.register(ops.ElementWiseVectorizedUDF)
+@pre_execute.register(ops.ElementWiseVectorizedUDF, ibis.client.Client)
+def pre_execute_elementwise_udf(op, *clients, scope=None, **kwargs):
+    """Register execution rules for elementwise UDFs.
+    """
+    input_type = op.input_type
+
+    # definitions
+
+    # Define an execution rule for elementwise operations on a
+    # grouped Series
+    nargs = len(input_type)
+
+    @execute_node.register(
+        ops.ElementWiseVectorizedUDF,
+        *(itertools.repeat(ddgb.SeriesGroupBy, nargs)),
+    )
+    def execute_udf_node_groupby(op, *args, **kwargs):
+        func = op.func
+
+        # all grouping keys must be identical
+        assert_identical_grouping_keys(*args)
+
+        # we're performing a scalar operation on grouped column, so
+        # perform the operation directly on the underlying Series
+        # and regroup after it's finished
+        args_objs = [make_selected_obj(arg) for arg in args]
+        groupings = args[0].index
+        return dd.map_partitions(func, *args_objs).groupby(groupings)
+
+    # Define an execution rule for a simple elementwise Series
+    # function
+    @execute_node.register(
+        ops.ElementWiseVectorizedUDF, *(itertools.repeat(pandas.Series, nargs))
+    )
+    @execute_node.register(
+        ops.ElementWiseVectorizedUDF, *(itertools.repeat(dd.Series, nargs))
+    )
+    @execute_node.register(
+        ops.ElementWiseVectorizedUDF, *(itertools.repeat(object, nargs))
+    )
+    def execute_udf_node(op, *args, **kwargs):
+        # We have rewritten op.func to be a closure enclosing
+        # the kwargs, and therefore, we do not need to pass
+        # kwargs here. This is true for all udf execution in this
+        # file.
+        # See ibis.udf.vectorized.UserDefinedFunction
+        return op.func(*args)
```

Comment:
I think func takes pd.Series not dd.Series here?

Reply:
Let's add some assertion in the test case to verify input object types

Reply:
https://github.com/ibis-project/ibis/pull/2648#discussion_r592887643